prov/util/src/util_mem_monitor.c

```diff
@@ -81,9 +81,9 @@ void ofi_monitors_init(void)
 	uffd_monitor->init(uffd_monitor);
 	memhooks_monitor->init(memhooks_monitor);
 
-#if defined(HAVE_ELF_H) && defined(HAVE_SYS_AUXV_H)
+#if defined(HAVE_ELF_H) && defined(HAVE_SYS_AUXV_H) && ENABLE_MEMHOOKS_MONITOR
         default_monitor = memhooks_monitor;
-#elif HAVE_UFFD_UNMAP
+#elif HAVE_UFFD_UNMAP && ENABLE_UFFD_MONITOR
```

Comment:
Similarly, switch to just ENABLE_UFFD_MONITOR and fold other checks in, so we don't need to worry about duplicating the more complicated checks everywhere else.